src/components/userFeedback/UserFeedback.spec.tsx

```diff
@@ -0,0 +1,147 @@
+import { shallow, ShallowWrapper } from 'enzyme';
+import { UserFeedback, userFeedbackState, IUserFeedbackProps } from './UserFeedback';
+
+// Until Webpack provided plugins works with TS 2.0
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+
+describe('<UserFeedback>', () => {
+  const defaultTextColorClass: string = 'text-dark-grey';
+  const errorTextColorClass: string = 'text-red';
+
+  const generateProps = (feedbackText: string, state: string, constantClasses?: string, displayOnShow?: string): IUserFeedbackProps => {
+    return displayOnShow ? { feedbackText, state, constantClasses, displayOnShow } : { feedbackText, state, constantClasses };
+  };
+
+  const defaultProps: IUserFeedbackProps = generateProps('', '', '');
+  const defaultPropsWithDisplay: IUserFeedbackProps = generateProps('', '', '', 'block');
+
+  const getShallowOutput = (props: IUserFeedbackProps): ShallowWrapper<IUserFeedbackProps, any> => {
+    return shallow(<UserFeedback {...props} />);
+  };
+
+  describe('"Shallow rendered" component', () => {
+    describe('rendering', () => {
+      it('should render without errors', () => {
+        expect(() => getShallowOutput(defaultProps)).not.toThrow();
+        expect(() => getShallowOutput(defaultPropsWithDisplay)).not.toThrow();
+        expect(() => getShallowOutput(generateProps('', '', ''))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', userFeedbackState.VALID, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', userFeedbackState.WARNING, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', userFeedbackState.ERROR, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', userFeedbackState.VALID, 'm1', 'block'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', userFeedbackState.WARNING, 'm1', 'block'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', userFeedbackState.ERROR, 'm1', 'block'))).not.toThrow();
+      });
+    });
+
+    describe('text content', () => {
+      it('should contain the exact text passed to prop feedbackText', () => {
+        let testText = 'testing';
+        let wrongText: string;
+
+        expect(getShallowOutput(generateProps(testText, '', '')).text() === testText).toBe(true);
```

Comment:
You can also write `expect(getShallowOutput(generateProps(testText, '', '')).text()).toBe(testText);` which is a little more explicit.

Same comment for the other values expected below.